timemachine/md/smc.py

```diff
@@ -120,6 +122,158 @@ def accumulate_results(samples, indices, log_weights, incremental_log_weights):
     return trajs_dict
 
 
+def adaptive_sequential_monte_carlo(
```

Comment:
nit: opportunity to revert duplication in the future: could express non-adaptive SMC as a special case, if this accepts a function `select_next_lam` in signature. (non-adaptive `select_next_lam` gets the next entry in a list, CESS-adaptive `select_next_lam` is implemented as below, ...)

Reply:
Can be a deboggle case for later